src/main/java/picard/util/IntervalListTools.java

```diff
@@ -16,15 +16,92 @@
 
 import java.io.File;
 import java.text.DecimalFormat;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
 
 /**
- * Little class to aid working with interval lists.
+ * Performs various {@link IntervalList} manipulations.
+ *
+ * <h3>Summary</h3>
+ * This tool offers multiple interval list file manipulation capabilities, including: sorting,
+ * merging, subtracting, padding, and other set-theoretic operations. The default action
+ * is to merge and sort the intervals provided in the {@link #INPUT}s. Other options, e.g. interval subtraction, are
+ * controlled by the arguments.
+ * <br />
+ * Both {@link IntervalList} and VCF files are accepted as input. {@link IntervalList} should be denoted with the extension
+ * {@value htsjdk.samtools.util.IOUtil#INTERVAL_LIST_FILE_EXTENSION}, while a VCF must have one of {@value htsjdk.samtools.util.IOUtil#VCF_FILE_EXTENSION}, {@value htsjdk.samtools.util.IOUtil#COMPRESSED_VCF_FILE_EXTENSION},
+ * {@value htsjdk.samtools.util.IOUtil#BCF_FILE_EXTENSION}. When VCF file is used as input, each variant is translated into an using its reference allele or the END
+ * INFO annotation (if present) to determine the extent of the interval.
+ *
+ * {@link IntervalListTools} can also "scatter" the resulting interval-list into many interval-files. This can be useful
+ * for creating multiple interval lists for scattering an analysis over.
+ *
+ * <h3>Details</h3>
+ *  The IntervalList file format is designed to help the users avoid mixing references when supplying intervals and
+ *  other genomic data to a single tool. A SAM style header must be present at the top of the file. After the header,
+ *  the file then contains records, one per line in text format with the following
+ * values tab-separated:
+ * <pre>
+ * <ul>
+ * <li>Sequence name (SN)</li>
+ * <li>Start position (1-based)</li>
+ * <li>End position (1-based, end inclusive)</li>
+ * <li>Strand (either + or -)</li>
+ * <li>Interval name (ideally unique names for intervals)</li>
+ * </ul>
+ * </pre>
+ * The coordinate system is such that the first base in a sequence is at position 1, and both the start
+ * and the end positions are included in an interval.
```

Comment:
Should we provide the technical description of this approach?  (Just 1-based, closed-end, I think?)